rails-commenting.rb

```diff
@@ -7,56 +7,100 @@
 # FILE: app/controller/blog_posts_controller.rb
 
 # ---1)
-class BlogPostsController < ApplicationController
-  def index
-    # ---2)
-    @posts = BlogPost.all
-  end
 
-  # ---3)
-  def show
-    @post = BlogPost.find(params[:id])
-  end
+# --- 
+  # 1) There is a class of BlogPostController that is a child of the ApplicationController.
+  # 2) This class has a method called 'index' 
 
+  class BlogPostsController < ApplicationController
+    def index
+  
+  # ---2)
+  
+      # 1) @post is tied to requesting all the index data in the BlogPost database.
+      # 2) It allows us to view the index.
+  
+      @posts = BlogPost.all
+    end
+  
+  # ---3)
+  
+      # 1) There is a method called 'show'
+      # 2) it takes in the parameter of the :id key.
+      # 3) So this is a method that finds a specific blog post.
+  
+    def show
+      @post = BlogPost.find(params[:id])
+    end
+  
   # ---4)
-  def new
-    @post = BlogPost.new
-  end
-
-  def create
-    # ---5)
+  
+      # 1) There is a method called 'new'
+      # 2) This one posts a new blog post.
+  
+    def new
+      @post = BlogPost.new
+    end
+  
+      # 3) The one below this is a method called 'create'
+    def create
+      
+  # ---5)
+      
+      # 4) It creates new parameters for the database BlogPost.
+  
     @post = BlogPost.create(blog_post_params)
-    if @post.valid?
-      redirect_to blog_post_path(@post)
+      if @post.valid?
+        redirect_to blog_post_path(@post)
+      end
     end
-  end
-
-  def edit
-    # ---6)
-    @post = BlogPost.find(params[:id])
-  end
-
-  def update
-    @post = BlogPost.find(params[:id])
-    # ---7)
-    @post.update(blog_post_params)
-    if @post.valid?
-      redirect_to blog_post_path(@post)
+  
+    def edit
+  
+  # ---6)
+  
+      # 1) The method 'edit' finds different blog post by :id key.
+  
+      @post = BlogPost.find(params[:id])
     end
-  end
-
-  def destroy
-    @post = BlogPost.find(params[:id])
-    if @post.destroy
-      # ---8)
-      redirect_to blog_posts_path
+  
+    def update
+      @post = BlogPost.find(params[:id])
+  
+  # ---7)
+  
+      # 1) The method 'update' edits/updates the blog posts if there is one.
+  
+      @post.update(blog_post_params)
+      if @post.valid?
+        redirect_to blog_post_path(@post)
+      end
     end
-  end
-
+  
+    def destroy
+      @post = BlogPost.find(params[:id])
+      if @post.destroy
+        
+  # ---8)
+  
+      # 1) the method 'destroy' is used to destroy/delete a specific blog post by its id.
+  
+        redirect_to blog_posts_path
+      end
+    end
+  
   # ---9)
-  private
-  def blog_post_params
-    # ---10)
-    params.require(:blog_post).permit(:title, :content)
+  
+    # 1) This private method below is used to hid something. i have no idea.
```

Comment:
You can use resources in this section!  The private section is to make it so that the methods below it are confined to only this class and cannot be called anywhere else within the application.  This is important as we do not want the permitted parameters accessible in other places than what is called in this controller.